src/lib.rs

```diff
@@ -59,11 +59,27 @@ pub fn maybe_grow<R, F: FnOnce() -> R>(red_zone: usize, stack_size: usize, callb
 /// The closure will still be on the same thread as the caller of `grow`.
 /// This will allocate a new stack with at least `stack_size` bytes.
 pub fn grow<R, F: FnOnce() -> R>(stack_size: usize, callback: F) -> R {
+    // Measuring with cargo-llvm-lines revealed that `psm::on_stack::with_on_stack`
+    // was monomorphized 1552 times in rustc and was responsible for 1.5% of
+    // rustc's total llvm IR lines. That takes time for LLVM to process.
```

Comment:
This metric is probably very misleading, because `with_on_stack` is the base frame into which LLVM is able to inline the provided callback. `with_on_stack` is also super small itself (< dozen of instructions), so I have a hard time believing that by itself `with_on_stack` is a cause of the entirety of 1.5% of instructions.

With that in mind, making the callback dynamic only serves to prevent the inlining of the callback, and thus now all those closures end up having their own "category" or "line" in `cargo-llvm-lines` output. The improvement in perf/comptime still great, however.

Reply:
I removed the numbers from the comment.

Yes, the actual functions that cargo-llvm-lines spits out are misleading, but they are a good to know where you can even _start looking_. And the total amount of llvm-lines decreased by 295 363 lines, so it's a big win. And yes, a separate entry for `stacker::grow::{{closure}}` appeared.